trollpy/tests.py

```diff
@@ -0,0 +1,357 @@
+                         'password': 'password'}).follow()
+    csrf = testapp.get('/add_smack').html.find_all("input")
+    csrf = csrf[0].attrs['value']
+    response = testapp.post('/profile/amos',
+                            params={'password': 'jdf;a',
+                                    'email': 'password@pe.com',
+                                    'first_name': 'Jeremy',
+                                    'last_name': 'renner',
+                                    'csrf_token': csrf}, status=302).follow()
+    assert 'password@pe.com' in response.html.find_all('li')[2].text
+    assert 'Jeremy renner' in response.html.find_all('li')[1].text
+
+
+def test_smack_api_route(testapp, fill_the_db):
+    """Test smack_api to return kill_score json."""
+    response = testapp.get('/smack_api', status=200)
+    assert len(response.json) == 1
+
+
+def test_new_user(testapp):
+    """Test that adding a new user via registration page."""
+    user = {
+        'username': 'dude',
+        'password': 'password',
+        'first_name': 'dude',
+        'last_name': 'asdasd',
+        'email': 'goyour@skdgmsk',
+
+    }
+    html = testapp.post('/registration', user, status=302).follow().html
+    assert 'goyour@skdgmsk' in html.find_all('li')[2].text
+    assert 'dude asdasd' in html.find_all('li')[1].text
+
+
+def test_new_user_with_spaces(testapp):
+    """Test that adding a new user via registration page."""
+    user = {
+        'username': 'dude mclovin',
+        'password': 'password',
+        'first_name': 'dude',
+        'last_name': 'asdasd',
+        'email': 'goyour@skdgmsk2',
+
+    }
+    html = testapp.post('/registration', user, status=302).follow().html
+    assert 'goyour@skdgmsk2' in html.find_all('li')[2].text
+    assert 'dude asdasd' in html.find_all('li')[1].text
+
+
+def test_login_user_with_spaces(testapp):
+    """Test that adding a new user via registration page with spaces."""
+    user = {
+        'username': 'dude mclovin',
+        'password': 'password',
+        'first_name': 'dude',
+        'last_name': 'asdasd',
+        'email': 'goyour@skdgmsk2',
+    }
+    testapp.post('/registration', user, status=302).follow().html
+    response = testapp.post('/login',
+                            params={'username': 'dude mclovin',
+                                    'password': 'password'}).follow()
+    assert response.status_code == 200
+
+
+def test_add_smack(testapp, fill_the_db):
+    """Test that add_smack route shows form."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    response = testapp.get('/add_smack', status=200)
+    assert len(response.html.find_all('form')) == 1
+
+
+def test_add_smack_with_no_user(testapp, fill_the_db):
+    """Test that add_smack route shows 403."""
+    assert testapp.get('/add_smack', status=403)
+
+
+def test_add_smack_post(testapp, fill_the_db):
+    """Test adding smack to db."""
```

Comment:
Are you testing that a new smack statement gets added to the DB and can be retrieved from JSON?